### PR TITLE
feat: add service for open ai

### PR DESCRIPTION
**Pull request by davidsoderberg**

### CODE REVIEW

File: apps/api/src/app/notification-template/notification-template.controller.ts [hunk 1]
```diff
@@ -39,10 +39,13 @@ import { Roles } from '../auth/framework/roles.decorator';
...
 import { DataBooleanDto } from '../shared/dtos/data-wrapper-dto';
 import { CreateNotificationTemplateQuery } from './queries';
+import { GetAiMessage } from './usecases/get-ai-message/get-ai-message.usecase';
+import { GetAiMessageCommand } from './usecases/get-ai-message/get-ai-message.command';
+import { OpenAiService } from '../shared/services/openai/openai.service';
 
 @Controller('/notification-templates')
 @UseInterceptors(ClassSerializerInterceptor)
-@UseGuards(JwtAuthGuard)
+// @UseGuards(JwtAuthGuard)
```
Comment (p-fernandez) [inline apps/api/src/app/notification-template/notification-template.controller.ts:48]: Is this temporary?

File: apps/api/src/app/notification-template/notification-template.controller.ts [hunk 2]
```diff
@@ -183,6 +188,42 @@ export class NotificationTemplateController {
...
+        environmentId: user.environmentId,
+        organizationId: user.organizationId,
+        userId: user._id,
+        prompt: body.prompt,
+        channel: body.channel,
+      })
+    );
+  }
+
+  @Get('/test/ai')
```
Comment (p-fernandez) [inline apps/api/src/app/notification-template/notification-template.controller.ts:212]:
```suggestion
  @Get('/ai/test')
```

File: apps/api/src/app/shared/services/openai/openai.service.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import type { ChatGPTAPI } from 'chatgpt';
+
+@Injectable()
+export class OpenAiService implements OnModuleInit {
+  private _api: ChatGPTAPI;
+
+  async onModuleInit() {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const { ChatGPTAPI: chat } = await (eval(`import('chatgpt')`) as Promise<{
```
Comment (p-fernandez) [inline apps/api/src/app/shared/services/openai/openai.service.ts:11]: 😱

**Review by p-fernandez — APPROVED** [review]

File: apps/api/src/app/notification-template/notification-template.controller.ts (hunk 1 above)
Comment (davidsoderberg) [reply apps/api/src/app/notification-template/notification-template.controller.ts:48]: yes

**Pull request merged**